### PR TITLE
Config Documentation

### PR DESCRIPTION
**Pull request by jcrichlake**
## Adding Docs

This PR adds ADRs around our usage of configs and a readme for FAQs that pertain to config questions. 

## Issue

[SFTP config](https://github.com/orgs/CDCgov/projects/45/views/1?pane=issue&itemId=82926101&issue=CDCgov%7Ctrusted-intermediary%7C1082)

### CODE REVIEW

**Review by basiliskus — COMMENTED** [review]
Looks good, just left one question

File: adr/012-configs.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+## Context
+
+In order to enable the usage of partner-specific settings in the different parts of the app, we are going to store the settings
+in a config container in our Azure storage account. Each partner will have its own separate file within the container to minimize potential
+blast radius when changing settings.
+
+
+## Notes
+- config files should only contain non-secret values
+- secrets will use a consistent naming pattern based on the partner ID used in config (so we can dynamically assemble the key names in code)
```
[the remark this answers is not in the document]
Comment (pluckyswan) [reply adr/012-configs.md]: SECRETS.md contains the current naming convention for secrets.

[the remark this answers is not in the document]
Comment (pluckyswan) [reply adr/012-configs.md]: It might be helpful to link out to it

File: docs/configs.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# FAQ
+- We use the partner's organization name in ReportStream as the partner ID
+- Config files are the partner ID plus `.json`
+- Config keys in code are the partner ID
```
Comment (pluckyswan) [inline docs/configs.md:4]: Could an example of what that would look like be added?

File: adr/012-configs.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jcrichlake) [reply adr/012-configs.md]: This has been added to where the notes were moved which is in configs.md

File: docs/configs.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# FAQ
```
Comment (jherrflexion) [inline docs/configs.md:1]: Loving the FAQ section. Great idea!

File: adr/012-configs.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# 12. Partner configuration
+
+Date: 2024-12-12
+
+## Decision
+
+We will store partner config settings in an Azure container
```
Comment (GilmoreA6) [inline adr/012-configs.md:7]: nit: consider being more specific here, .i.e. "We will store partner sftp config settings..."

Comment (jcrichlake) [reply adr/012-configs.md]: Fixed, thanks!

File: docs/configs.md (hunk 2 above)
Comment (jcrichlake) [reply docs/configs.md:4]: Added

**Review by pluckyswan — APPROVED** [review]

**Pull request merged**